web/docs/language/features.md

```diff
@@ -178,138 +178,246 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
-However, that can often be tricky and error-prone, so Wasp offers quick and effective solution to get you started: automatic invalidation of query cache based on entities that queries / actions are using.
+// ...
 
-Specifically, if Action A1 uses Entity E1 and Query Q1 also uses Entity E1 and Action A1 is executed, Wasp will recognize that Q1 might not be up-to-date any more and will therefore invalidate its cache, making sure it gets updated.
+sayHi()
+```
 
-In practice, this means that without really even thinking about it, Wasp will make sure to keep the queries up to date for you in regard with the changes done by actions.
 
-On the other hand, this kind of automatic invalidation of cache can be wasteful (updating when not needed) and will not work if other resources than entities are used. In that case, make sure to use mechanisms provided by react-query for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
+More differences and Action/Query specific features will come in future versions of Wasp.
 
+### Cache Invalidation
+One of the trickiest parts of managing a web app's state is making sure the data shown by the queries is up to date.
+Since Wasp uses _react-query_ for Query management, we must make sure to invalidate its caches whenever they become stale.
 
-### Action
+It's possible to invalidate the caches manually through several mechanisms _react-query_ provides (e.g., refetch, direct invalidation).
+However, since manual cache invalidation quickly becomes complex and error-prone, Wasp offers a quicker and a more effective solution to get you started: **automatic Entity-based Query cache invalidation**.
+Because Actions can (and most often do) modify the state while Queries read it, Wasp invalidates a Query's cache whenever an Action that uses the same Entity is executed.
+
+For example, let's assume that Action `A1` and Query `Q1` both use Entity `E1`. If `A1` is executed, `Q1`'s cached result may no longer be up-to-date.
+Wasp will therefore invalidate it, making `Q1` up to date again.
 
-Actions are very similar to Queries, so similar that we will only list the differences:
-1. They can modify state (queries can't).
-2. There is no special React hook for them (like `useQuery` for Queries), you just call them directly.
-3. They are declared in Wasp in same way as Queries, but keyword is `action`, not `query`.
+In practice, this means that Wasp keeps the queries "fresh" without requiring you to think about cache invalidation.
 
-More differences and action/query specific features will come in the future versions of Wasp.
+On the other hand, this kind of automatic cache invalidation can become wasteful (some updates might not be necessary) and will not only work for Entities. If that's an issue, you can use the mechanisms provided by _react-query_ for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
```

Comment:
```suggestion
On the other hand, this kind of automatic cache invalidation can become wasteful (some updates might not be necessary) and will only work for Entities. If that's an issue, you can use the mechanisms provided by _react-query_ for now, and expect more direct support in Wasp for handling those use cases in a nice, elegant way.
```
Did you mean it like this? Although still not completely sure what you meant to say here?

Reply:
It's a mistake, nice catch. Leftover "not" from one of the previous iterations.